group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -8120,56 +8119,58 @@ public Map.Entry<AlterShareGroupOffsetsResponseData, InitializeShareGroupStatePa
         Map<Uuid, Map<Integer, Long>> offsetByTopicPartitions = new HashMap<>();
 
         alterShareGroupOffsetsRequest.topics().forEach(topic -> {
-            TopicImage topicImage = metadataImage.topics().getTopic(topic.topicName());
-            if (topicImage != null) {
-                Uuid topicId = topicImage.id();
-                Set<Integer> existingPartitions = new HashSet<>(topicImage.partitions().keySet());
-                List<AlterShareGroupOffsetsResponseData.AlterShareGroupOffsetsResponsePartition> partitions = new ArrayList<>();
-                topic.partitions().forEach(partition -> {
-                    if (existingPartitions.contains(partition.partitionIndex())) {
-                        partitions.add(
-                            new AlterShareGroupOffsetsResponseData.AlterShareGroupOffsetsResponsePartition()
-                                .setPartitionIndex(partition.partitionIndex())
-                                .setErrorCode(Errors.NONE.code()));
-                        offsetByTopicPartitions.computeIfAbsent(topicId, k -> new HashMap<>()).put(partition.partitionIndex(), partition.startOffset());
-                    } else {
-                        partitions.add(
-                            new AlterShareGroupOffsetsResponseData.AlterShareGroupOffsetsResponsePartition()
-                                .setPartitionIndex(partition.partitionIndex())
-                                .setErrorCode(Errors.UNKNOWN_TOPIC_OR_PARTITION.code())
-                                .setErrorMessage(Errors.UNKNOWN_TOPIC_OR_PARTITION.message()));
-                    }
-                });
-
-                initializingTopics.put(topicId, new InitMapValue(
-                    topic.topicName(),
-                    topic.partitions().stream()
-                        .map(AlterShareGroupOffsetsRequestData.AlterShareGroupOffsetsRequestPartition::partitionIndex)
-                        .filter(existingPartitions::contains)
-                        .collect(Collectors.toSet()),
-                    currentTimeMs
-                ));
+            metadataImage.topicMetadata(topic.topicName()).ifPresentOrElse(
```

Comment:
This isn't really the ifPresentOrElse's fault but I do find these if blocks a little unwieldy. I don't really have a great suggestion right now -- it's probably better that we leave the current structure so that the refactoring is still fairly mechanical.

Reply:
Updated it to be a regular `if/else` block